scale/gcloud_update.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/python
+
+import subprocess
+from settings import GCLOUD_INSTANCE_GROUP, GCE_ZONE
+from utils import getPods, getPodType, getPodNamespace,\
+    get_singleuser_image_value
+
+import logging
+
+
+def shutdownSpecifiedNode(name):
+    """Deletes the specified node by calling the Google Cloud Engine"""
+    cmd = ['gcloud', 'compute', 'instance-groups', 'managed', 'delete-instances',
+           GCLOUD_INSTANCE_GROUP, '--instances=' + name]
+    p = subprocess.Popen(' '.join(cmd), stdout=subprocess.PIPE, shell=True)
```

Comment:
`shell=True` should *never* be used in any code - it's usually just a security disaster waiting to happen. https://stackoverflow.com/questions/3172470/actual-meaning-of-shell-true-in-subprocess and others:

> Unlike some other popen functions, this implementation will never implicitly call a system shell. This means that all characters, including shell metacharacters, can safely be passed to child processes. If the shell is invoked explicitly, via shell=True, it is the application’s responsibility to ensure that all whitespace and metacharacters are quoted appropriately to avoid shell injection vulnerabilities.
> From https://docs.python.org/3/library/subprocess.html#security-considerations

in this case, anyone who can control the value of the `name` parameter now has arbitrary shell execution rights on wherever this script is running (since they can just set name to '; rm -rf /'). 

I'd suggest:

1. Getting path to gcloud explicitly
2. Passing it as a list 
3. Using `subprocess.check_output`, which does all the things you want (rather than use popen directly)